sdk/appconfiguration/azure-spring-cloud-feature-management/README.md

```diff
@@ -259,6 +259,104 @@ Options are available to customize how targeting evaluation is performed across
+        configuration-reference: "DiscountBanner:Big"
+        assignment-parameters:
+          Audience:
+            users:
+              - Jeff
+              - Alicia
+            groups:
+              -
+                 name: "Ring0"
+                 rolloutPercentage: 80
+              -
+                 name: "Ring1"
+                 rolloutPercentage: 50
+            defaultRolloutPercentage: 20
+      -
+        name: Small
+        default: false
+        configuration-reference: "DiscountBanner:Small"
+        assignment-parameters:
+          Audience:
+            users:
+              - "Jack"
+              - "Amanda"
+            groups:
+              -
+                 name: "Ring0"
+                 rolloutPercentage: 10
+              -
+                 name: "Ring1"
+                 rolloutPercentage: 2W0
+            defaultRolloutPercentage: 0
+feature-variant:
+  DiscountBanner:
+    Big:
+      size: 400
+      color: "#BBB"
+    Small:
+      size: 150
+      color: "#DDD"
+```
+
+A Dynamic Feature has two main parts the assigner and a list of variants. The assigner in this case `feature-management.DiscountBanner.assigner` is pointing the `FeatureFilter` `Microsoft.Targeting` that will evaluate the list of variants and choose which configuration-reference will be used. Each variant has; a name, configuration-reference, and assignment-parameters, and one of them needs to be set as the default variant. The name field is just meta data to reference the variant and just needs to be unique. The configuration-reference is a link to the configurations defined under `feature-variant` in the configuration file. **Note:** the colon notation is required when denoting parts of the configuration. If it is needed multiple colons can be used to reference sub configurations. i.e. `DiscountBanner:Big:size` would return the `Integer` 400.
+
+The assignment-parameters work similarly to the other `FeatureFilters`. It can take in a `Map` which contains the evaluation criteria. In the case of `Microsoft.Targeting` it contains the `Audience` object. This enables custom assigners to be created.
+
+The `Microsoft.Targeting` assigner uses three evaluation criteria to assign a variant; users, groups, and defaultRolloutPercentage. Assignment requires the definition of the `ITargetingContextAccessor` where a user defines how the user and group parameters are defined.
+
+If a users is listed in a variant then they are automatically given that variant. If they are in more than one variant they are returned to the first listed variant.
+
+Groups have a rollout percentage. Using the user and group information users in groups are bucketed into parts of the rollout. This enables the a rollout where 10 of a group gets a variant, then when it increases to 50% that 10% still gets the variant with an additional 40% of the group now getting the variant. The rolloutPercentage for a group is shared across all variants that use that group.
```

Comment:
"This enables the a rollout" should be "This enables a rollout"

Reply:
This paragraph is hard to follow. Also, don't we define targeting semantics elsewhere in such a way that we can reference them and just explain how targeting assignment builds on top of it?